### ISSUE TITLE
HTTP Requests with Tinygo

### ISSUE THREAD

**Issue by Nv7-GitHub** [vugu/vugu#182]
There is vugu/vjson for JSON support in tinygo, but there isn't anything for sending requests. It would be great to have http.Get, perhaps through the `fetch` API?

**Comment by bradleypeabody** [comment]
Yeah I agree - that's what the main Go project's net/http implementation does on js + wasm.

See https://github.com/tinygo-org/tinygo/issues/850

**Comment by Nv7-GitHub** [comment]
When I try to use the `fetch` I get a go deadlock error, how can you properly wait for a `fetch` call without using callbacks? The Go standard library `net/http` doesn't cause a deadlock, but using `fetch` with a handler and passing the result into a channel does cause a deadlock, is there anyway to get around this?

**Comment by Nv7-GitHub** [comment]
Someone already did it! I'm trying out https://github.com/marwan-at-work/wasm-fetch

**Comment by gschleic** [comment]
I tried it out but getting:
```
Unhandled Promise Rejection: LinkError: import function env:time.stopTimer must be callable
```
Calling from embedded WASM on browser compiled with  tinygo version 0.24.0

**Comment by bradleypeabody** [comment]
Yeah I think the issue here is TinyGo generally doesn't have a good way to implement net/http on Microcontrollers (TinyGo's original and primary target environments) because the environment is just so different - network devices in an embedded scenario have radically different APIs, even between each other, and are highly hardware dependent.   So a key question here is if the TinyGo team would be open to the idea of having an WASM-only implementation of net/http.  If so, that is likely the best place for it.  Otherwise something like what was done for vjson could be done for this, which would basically involve extracting and hacking the net/http WASM implementation from the mainline Go language and seeing if it can be mangled to work in TinyGo (I suspect it can).

**Comment by owenwaller** [comment]
This seems much more a TinyGo issue than a Vugu issue, if it hasn't been resolved in the current versions of TinyGo.

I've marked this as an improvement, post v1.0 release, in common with all the other TinyGo issues

**Comment by Nv7-GitHub** [comment]
One problem I had with Vugu was a very large binary which greatly hindered website loading performance and I couldn't switch to TinyGo and resolve this due to this issue, so I was thinking that this should be a major priority. How are you dealing with this problem?

**Comment by owenwaller** [comment]
@Nv7-GitHub 

Thanks for the comment. At the minute the tentative plan is to get the code in better shape for v1.0 and then to look at using TinyGo.

Neither @bradleypeabody or myself are up to date with current state of TinyGo. There are some tests that should (or at least did) run under TinyGo, but against a older version. The latest TinyGo is v0.30.0 but the tests look like they are against v0.9.0.

But I am about to open a new issue specific to TinyGo.

**Comment by owenwaller** [comment]
@Nv7-GitHub see #251

**Comment by owenwaller** [comment]
See also #268
